$OpenBSD: patch-src_hexcurse_c,v 1.1 2019/11/17 21:03:29 fcambus Exp $

Show help if no filename argument is supplied, otherwise it's not possible
to quit the editor.

Upstream commit 37f94f2c1d0a3cc5110b2771e02ed894522d65d3.

Index: src/hexcurse.c
--- src/hexcurse.c.orig
+++ src/hexcurse.c
@@ -217,9 +217,13 @@ off_t parseArgs(int argc, char *argv[])
         fpINfilename = strdup(argv[0]);
     }
 
-    if (fpINfilename && strcmp(fpINfilename, ""))
+    if (fpINfilename == NULL) {
+        print_usage();
+        exit(-1);
+    } else if (fpINfilename && strcmp(fpINfilename, "")) {
         if ((fpIN = fopen(fpINfilename, "r")) == NULL)
             exit_err("Could not open file");
+    }
 
     return ((fpIN != NULL) ? maxLoc(fpIN):0);		/* return file length */
 }
